blocksync/errors.go

```diff
@@ -0,0 +1,49 @@
+package blocksync
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/gogoproto/proto"
+)
+
+var (
+	// ErrNilMessage is returned when provided message is empty
+	ErrNilMessage = errors.New("message cannot be nil")
+)
+
+// ErrInvalidBase is returned when peer informs of a status with invalid height
+type ErrInvalidHeight struct {
+	Height int64
+	Reason string
+}
+
+func (e ErrInvalidHeight) Error() string {
+	return fmt.Sprintf("Invalid height %v. %s", e.Height, e.Reason)
```

Comment:
```suggestion
	return fmt.Sprintf("invalid height %v: %s", e.Height, e.Reason)
```

Errors might be nested.